app/src/features/surveys/view/SurveyObservations.tsx

```diff
@@ -186,6 +193,42 @@ const SurveyObservations = () => {
     </IconButton>
   );
 
+  const groupType = {
+    mandatory: 'Mandatory fields have not been filled out in your file',
+    unsupported_header: 'Column headers in your file are not supported',
+    miscellaneous: 'Miscellaneous'
+  };
+
+  const messageList = submissionStatus?.messages;
+
+  const messageObject =
+    (messageList &&
+      messageList.reduce((workingData: any, row: any) => {
+        let group;
+
+        switch (row.error_code) {
+          case 'missing_required_header':
+          case 'missing_required_field':
```

Comment:
question:
is this how you do it if multiple cases have the same logic? or can we remove line 210?